dashboard/test/ui/features/hourOfCode_signedIn.feature

```diff
@@ -57,20 +52,27 @@ Scenario: Failing at puzzle 6, refreshing puzzle 6, bubble should show up as att
   And I debug cookies
   When element "#runButton" is visible
   And I debug cookies
-  Then element ".progress_container div:nth-child(6) a" has class "level_link attempted"
+  Then I verify progress in the header of the current page is "attempted" for level 6
 
 Scenario: Async progress write followed by a stale read
   Given I am on "http://studio.code.org/hoc/20?noautoplay=true"
-  And element ".header_middle a.level_link:first" does not have class "perfect"
+  And I verify progress in the header of the current page is "not_tried" for level 20
   Then mark the current level as completed on the client
   And I reload the page
-  And I wait to see ".header_middle"
-  And I wait for 2 seconds
-  And element ".header_middle a.level_link:last" has class "perfect"
-  Then I am on "http://studio.code.org/s/hourofcode"
-  And I wait to see ".user-stats-block"
-  And I wait for 2 seconds
-  And element ".user-stats-block .games:first a.level_link:last" has class "perfect"
+  And I verify progress in the header of the current page is "perfect" for level 20
+  And I navigate to the course page and verify progress for course "hourofcode" stage 1 level 20 is "perfect"
+
+Scenario: Progress on the server that is not on the client
+  Given I am on "http://studio.code.org/hoc/20?noautoplay=true"
+  And I verify progress in the header of the current page is "not_tried" for level 20
+  And I close the dialog
+  And I wait until element "#runButton" is visible
+  And I press "runButton"
+  And I wait to see ".modal"
+  Then I am on "http://studio.code.org/hoc/reset"
```

Comment:
Nit: I'd still like a comment or something to the effect that this navigation step resets progress in local storage, but we still expect to get progress from the server.